### ISSUE TITLE
Monitoring multiple subreddits in a foreach? The key was not present in the dictionary

### ISSUE THREAD

**Issue by Marcel0024** [sirkris/Reddit.NET#98]
I have this:

    _logger.LogInformation($"Started bot in {_env.EnvironmentName}");

    foreach (var subredditToMonitor in _monitorSettings.Subreddits)
    {
        _logger.LogInformation($"Started monitoring {subredditToMonitor.Name}");

        var subreddit = _redditClient.Subreddit(subredditToMonitor.Name);

        subreddit.Comments.GetNew();
        subreddit.Comments.MonitorNew();
        subreddit.Comments.NewUpdated += C_NewCommentsUpdated;
    }

Basically i just wanna monitor multiple subreddits for new comments.

I get this:
![image](https://user-images.githubusercontent.com/14852157/72736696-7ad6f200-3b9e-11ea-9baa-d328ef76d6fe.png)

Without a foreach it runs fine. How can i accomplish my goal?

EDIT: It's for this baby right here: https://github.com/Marcel0024/RedditBots/blob/c1b3def1599c79c4685eab4a525a0a11b474c1e2/src/Bots/PapiamentoBot.cs#L46

**Comment by Marcel0024** [comment]
Looking at the code, is it because all new monitoring threads are created with a key like "NewComments", therefor starting a new thread for a different subreddit, it tries to create a thread with the same key?

**Comment by Marcel0024** [comment]
So i helped myself and dug in. I found the bug. 

So the 
`  var subreddit = _redditClient.Subreddit(subredditToMonitor.Name);`

This is in the constructor of Subreddit (method called above):

https://github.com/sirkris/Reddit.NET/blob/master/src/Reddit.NET/Controllers/Subreddit.cs#L388
Note only 2 parameters when calling new Comments(1, 2);

This is the constructor of Comments:
https://github.com/sirkris/Reddit.NET/blob/master/src/Reddit.NET/Controllers/Comments.cs#L368

Because it's coming from Subreddit the FullName is always NULL. And if you noted from above the PostId is also NULL from the Comments constructor. 
Therefor the SubKey is always "t3_". Hence the duplicate key exception.

Will be submitting a PR shortly

**Comment by sirkris** [comment]
Merged into 1.4.
